README.md

```diff
@@ -2,48 +2,58 @@ franz-go
 ===
 
 [![GoDev](https://img.shields.io/static/v1?label=godev&message=reference&color=00add8)][godev]
+![GitHub](https://img.shields.io/github/license/twmb/franz-go)
+![GitHub tag (latest SemVer)](https://img.shields.io/github/v/tag/twmb/franz-go)
+[![Discord Chat](https://img.shields.io/badge/discord-online-brightgreen.svg)](https://discord.gg/K4R5c8zsMS)
 
 [godev]: https://pkg.go.dev/github.com/twmb/franz-go/pkg/kgo
 
-## Contents
+Franz-go is an all-encompassing Kafka client fully written Go. This library aims to provide **every Kafka feature** from 
+Apache Kafka v0.8.0 onward. It has support for transactions, regex topic consuming, the latest partitioning strategies,
+data loss detection, closest replica fetching, and more. If a client KIP exists, this library aims to support it.
 
-- [Introduction](#introduction)
-- [Stability Status](#stability-status)
-- [Performance](#performance)
-- [Record Reliability](#record-reliability)
-- [EOS](#eos)
-- [Consumer Groups](#consumer-groups)
-    - [Offset Management](#offset-management)
-        - [Without Transactions](#without-transactions)
-        - [Within Transactions](#within-transactions)
-    - [The Cooperative Balancer](#the-cooperative-balancer)
-    - [Static Members](#static-members)
-- [Version Pinning](#version-pinning)
-- [General Admin Requests](#general-admin-requests)
-- [TLS](#tls)
-- [Logging](#logging)
-- [Metrics](#metrics)
-- [Supported KIPs](#supported-kips)
+This library attempts to provide an intuitive API while interacting with Kafka the way Kafka expects (timeouts, etc.).
 
-## Introduction
+## Getting started
 
-franz-go is an all-encompassing Kafka consumer api in Go. The main package is kgo,
-but kmsg also exists containing autogenerated code for every request and
-response type, and a few other miscellaneous packages exist.
+Basic usage looks like this:
 
-The main package kgo is a high performance, pure Go library for interacting
-with Kafka. This library aims to provide every Kafka feature from 0.8.0+.
+```go
```

Comment:
I think the most common / most expected usage of this package would be for producing or consuming, so it may be better to have one message produced here (maybe the callback can use a wait group just to show), and then consuming that same message.

Maybe something like

```go
ctx := context.Background()

// All record production goes through Produce, and the callback can be used
// to allow for syncronous or asyncronous production.
var wg sync.WaitGroup
wg.Add(1)
record := &kgo.Record{Topic: "foo", Value: []byte("bar")}
err := client.Produce(ctx, record, func(_ *Record, err error) {
        defer wg.Done()
        if err != nil {
                fmt.Printf("record had a produce error: %v\n", err)
        }

}
if err != nil {
        panic("we are unable to produce if the context is canceled, we have hit max buffered," +
                "or if we are transactional and not in a transaction")
}
wg.Wait()

// Consuming can either be direct (no group), or through a group.
// Below, we use a group.
client.AssignGroup("group",
        kgo.GroupTopics("foo"))

for {
        fetches := client.PollFetches(ctx)
        // process fetches
}
```